socs/agents/srs_cg635/agent.py

```diff
@@ -0,0 +1,237 @@
+        self.log.info("Clearing event registers and error queue.")
+        self.clock.clear()
+        return True
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    @ocs_agent.param('wait', default=1, type=float)
+    def acq(self, session, params):
+        """acq(wait=1, test_mode=False)
+
+        **Process** - Continuously monitor srs clock lock registers
+        and publish to a feed.
+
+        The ``session.data`` object stores the most recent published values
+        in a dictionary. For example::
+
+            session.data = {
+                'timestamp': 1598626144.5365012,
+                'block_name': 'clock_output',
+                'data': {
+                    'Frequency': 122880000.0
+                    'Standard_CMOS_Output': 3,
+                    'Running_State': 1,
+                    'Timebase': 3,
+                    'PLL_RF_UNLOCKED': 1,
+                    'PLL_19MHZ_UNLOCKED': 1,
+                    'PLL_10MHz_UNLOCKED': 0,
+                    'PLL_RB_UNLOCKED': 1,
+                    'PLL_OUT_UNLOCKED': 0,
+                    'PLL_Phase_Shift': 0,
+                }
+            }
+
+        Refer to drivers for interpretation of outputs.
+
+        Parameters
+        ----------
+        wait: float, optional
+            time to wait between measurements [seconds]. Default=1s.
+
+        """
+        self.monitor = True
+
+        while self.monitor:
+            with self.lock.acquire_timeout(1) as acquired:
+                if acquired:
+                    if not self.clock:
+                        self.log.info("Trying to reconnect...")
+                        self._initialize_interface()
+                        continue
+
+                    data = {
+                        'timestamp': time.time(),
+                        'block_name': 'clock_output',
+                        'data': {}
+                    }
+
+                    try:
+                        data['data']['Frequency'] = self.clock.get_freq()
+                        data['data']['Standard_CMOS_Output'] = self.clock.get_stdc()
+                        data['data']['Running_State'] = self.clock.get_runs()
+                        data['data']['Timebase'] = self.clock.get_timebase()
+
+                        # get_lock_statuses returns a dict of the register bits
+                        # Loop through the items to add each to the data
+                        lock_statuses = self.clock.get_lock_statuses()
+                        for register, status in lock_statuses.items():
+                            # Not adding PLL causes a naming error
+                            # Two of the registers start with an number
+                            data['data']["PLL_" + register] = status
+
+                    except socket.timeout as e:
+                        self.log.error(f"Timeout in retrieving clock data: {e}")
+                        self.clock = None
```

Comment:
Add `session.degraded = True` here too:
```suggestion
                        self.clock = None
                        session.degraded = True
```